Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Application/SignContract/SignContractCommandHandler.cs

```diff
@@ -9,22 +8,21 @@ internal sealed class SignContractCommandHandler(
     IContractsRepository contractsRepository,
     IBindingContractsRepository bindingContractsRepository,
     TimeProvider timeProvider,
-    IPublishEndpoint publishEndpoint) : IRequestHandler<SignContractCommand, Guid>
+    IPublishEndpoint publishEndpoint) : IRequestHandler<SignContractCommand, ErrorOr<Guid>>
 {
-    public async Task<Guid> Handle(SignContractCommand command, CancellationToken cancellationToken)
-    {
-        var contract = await contractsRepository.GetByIdAsync(command.Id, cancellationToken) ??
-                       throw new ResourceNotFoundException(command.Id);
-        var now = timeProvider.GetUtcNow();
-        var bindingContract = contract.Sign(command.SignedAt, now);
-        await bindingContractsRepository.AddAsync(bindingContract, cancellationToken);
-        await contractsRepository.CommitAsync(cancellationToken);
-        var @event = ContractSignedEvent.Create(contract.Id.Value,
-                                                        contract.CustomerId,
-                                                        contract.SignedAt!.Value,
-                                                        contract.ExpiringAt!.Value);
-        await publishEndpoint.Publish(@event, cancellationToken);
+    public async Task<ErrorOr<Guid>> Handle(SignContractCommand command, CancellationToken cancellationToken) =>
+        await contractsRepository.GetByIdAsync(command.Id, cancellationToken)
+            .ThenAsync(async contract => await contract.Sign(command.SignedAt, timeProvider.GetUtcNow())
+                .ThenAsync(async bindingContract =>
```

Comment:
Move that it is in line with previous ThenAsync, it is easier to read it

Reply:
When i use autoformat rider format in that way. I would keep it like it is, beacuse when some run autoformat on solution will change formating. It is acceptable for you?